cmd/spiffe-helper/main.go

```diff
@@ -23,7 +23,30 @@ func main() {
 	flag.Parse()
 	log := logrus.WithField("system", "spiffe-helper")
 
-	if err := startSidecar(*configFile, *daemonModeFlag, log); err != nil {
+	log.Infof("Using configuration file: %q", *configFile)
+	hclConfig, err := config.ParseConfig(*configFile, *daemonModeFlag, daemonModeFlagName)
+	if err != nil {
+		log.WithError(err).Errorf("failed to parse configuration")
+		os.Exit(1)
+	}
+
+	if err := hclConfig.ValidateConfig(log); err != nil {
+		log.WithError(err).Errorf("invalid configuration")
+		os.Exit(1)
+	}
+
+	sidecarConfig := config.NewSidecarConfig(hclConfig, log)
+	spiffeSidecar := sidecar.New(sidecarConfig)
+
+	if *hclConfig.DaemonMode && hclConfig.HealthCheck.ListenerEnabled {
+		if err := health.StartHealthServer(hclConfig.HealthCheck, log, spiffeSidecar); err != nil {
+			log.WithError(err).Errorf("Error starting spiffe-helper health check server")
+			os.Exit(1)
+		}
+	}
```

Comment:
That looks better now. Last bit of cleanup, can you move this block to `startSidecar()`? We should be good to go after that.